### PR TITLE
FlxText: adding proper FlxTextAlign <----> TextFormatAlign conversion fu...

### PR DESCRIPTION
**Pull request by larsiusprime**
...nctions that work for both legacy and next

### CODE REVIEW

**Comment by larsiusprime** [comment]
Note this depends on this PR as well:
https://github.com/HaxeFlixel/flixel/pull/1474

File: flixel/text/FlxText.hx [hunk 1]
```diff
@@ -1066,4 +1048,32 @@ abstract FlxTextAlign(String) from String
...
 	var RIGHT = "right";
 	var JUSTIFY = "justify";
+	
+	#if !openfl_legacy
+	public static function fromOpenFL(tf:TextFormatAlign):FlxTextAlign
+	#else
+	public static function fromOpenFL(tf:String):FlxTextAlign
+	#end
+	{
+		//ifs instead of a switch b/c you can only switch on lower-case variable names and TextFormatAlign are all caps constants in legacy
```
Comment (Gama11) [inline flixel/text/FlxText.hx]: !?

Comment (larsiusprime) [inline flixel/text/FlxText.hx]: See here:
https://github.com/HaxeFlixel/flixel-ui/issues/99#issuecomment-92924244

I tried it initially with a switch function, but when compiling in legacy mode it wouldn't work.

**Comment by MSGhero** [comment]
I was wondering when this issue would come up in flixel. I don't like that openfl uses a straight enum, as even haxe's flash.text.TFA is an (extern) abstract enum. But oh well, nice change.

**Comment by larsiusprime** [comment]
So what's the consensus? Pull this in or convince Joshua to change TextFormatAlign to an abstract string enum?

**Comment by Beeblerox** [comment]
let's merge it

File: flixel/text/FlxText.hx (hunk 1 above)
Comment (Gama11) [inline flixel/text/FlxText.hx]: Define "trouble" (error message?), seems to be working here.

Comment (larsiusprime) [inline flixel/text/FlxText.hx]: Hmm, you're right, now it's suddenly working again. 

The error *was* "Capture variables must be lower-case"

**Pull request merged**